File: src/openstaad_mcp/staad_skills/staad-reports/scripts/create-force-report.py
"""Create a report table summarizing member end forces."""
geo = staad.Geometry
out = staad.Output
tbl = staad.Table
load = staad.Load

if not out.AreResultsAvailable():
    print("No results available - run analysis first")
else:
    beam_list = list(geo.GetBeamList())
    lc_numbers = list(load.GetPrimaryLoadCaseNumbers())
    lc = lc_numbers[0]

    report_no = tbl.CreateReport("Member Forces Summary")
    table_no = tbl.AddTable(report_no, "End Forces LC " + str(lc), len(beam_list), 4)

    # Headers
    headers = ["Member", "FX", "FY", "MZ"]
    for i, h in enumerate(headers, 1):
        tbl.SetColumnHeader(report_no, table_no, i, h)
        tbl.SetCellTextBold(report_no, table_no, 1, i)

    # Data
    for row, bid in enumerate(beam_list, 1):
        f = out.GetMemberEndForces(bid, 0, lc, 0)
        tbl.SetCellValue(report_no, table_no, row, 1, str(bid))
        tbl.SetCellValue(report_no, table_no, row, 2, f"{f[0]:.3f}")
        tbl.SetCellValue(report_no, table_no, row, 3, f"{f[1]:.3f}")
        tbl.SetCellValue(report_no, table_no, row, 4, f"{f[5]:.3f}")

    tbl.SetCellTextSizeAll(report_no, table_no, 10.0)
    tbl.SaveReport(report_no)
    print(f"Report created with {len(beam_list)} rows")

File: src/openstaad_mcp/staad_skills/staad-reports/scripts/create-force-report.js
// create-force-report.js
// Creates a report table summarizing member end forces.

const geo = staad.Geometry;
const out = staad.Output;
const tbl = staad.Table;
const load = staad.Load;

if (!out.AreResultsAvailable()) {
    console.log('No results available - run analysis first');
} else {
    const beamList = geo.GetBeamList();
    const lcNumbers = load.GetPrimaryLoadCaseNumbers();
    const lc = lcNumbers[0];

    const reportNo = tbl.CreateReport('Member Forces Summary');
    const tableNo = tbl.AddTable(reportNo, `End Forces LC ${lc}`, beamList.length, 4);

    // Headers
    const headers = ['Member', 'FX', 'FY', 'MZ'];
    headers.forEach((h, i) => {
        const col = i + 1;
        tbl.SetColumnHeader(reportNo, tableNo, col, h);
        tbl.SetCellTextBold(reportNo, tableNo, 1, col);
    });

    // Data
    beamList.forEach((bid, idx) => {
        const row = idx + 1;
        const f = out.GetMemberEndForces(bid, 0, lc, 0);
        tbl.SetCellValue(reportNo, tableNo, row, 1, String(bid));
        tbl.SetCellValue(reportNo, tableNo, row, 2, f[0].toFixed(3));
        tbl.SetCellValue(reportNo, tableNo, row, 3, f[1].toFixed(3));
        tbl.SetCellValue(reportNo, tableNo, row, 4, f[5].toFixed(3));
    });

    tbl.SetCellTextSizeAll(reportNo, tableNo, 10.0);
    tbl.SaveReport(reportNo);
    console.log(`Report created with ${beamList.length} rows`);
}
